src/deploy/osp_deployer/director.py

```diff
@@ -408,6 +414,76 @@ def setup_templates(self):
         self.setup_environment()
         self.setup_sanity_ini()
 
+    def clamp_min_pgs(self, num_pgs):
+        pg_options = [8192, 4096, 2048, 1024, 512, 256, 128, 64, 32, 16, 8, 4,
+                      2, 1]
+        pg_max = pg_options[0]
+
+        option_index = 0
+        finding_pg_value = True
+        while finding_pg_value:
+            if num_pgs >= pg_options[option_index]:
+                num_pgs = pg_options[option_index]
+                finding_pg_value = False
+            else:
+                option_index += 1
+
+        return num_pgs
+
+    def calc_pgs(self, num_osds, num_heavy_pools, num_other_pools):
+        replication_factor = 3
+        max_pgs = num_osds * 200 / replication_factor
```

Comment:
I'm puzzled seeing replication factor here.  How does that fit into calculating PGs?

Reply:
I *think* that was in your original algorithm.  If you go to the following link and search for "Logic behind Suggested PG Count", there's a formula that talks about calculating the number of pgs that includes dividing by the replication factor: https://ceph.com/pgcalc/

Reply:
The relation between PGs value and replication is very loose/indirect.  That's why I was surprised to see it.  I'm ok with it being there.  The logic to clamp the values is solid and would be able to resolve weird cases involving non-standard replica values (e.g., 2 is now common for all NVMe).